### PR TITLE
feat(Content): introduce Content component (replacement for Text)

### LINKED ISSUE #10395: Change Text component name to Content to better align with core
**Describe the enhancement or change**
Currently the component which is named Text in React uses `content` as its class name in core, which can cause confusion.

Additionally in #10390 we will be moving the component in a direction where Text as a component name isn't ideal.

For those reasons I believe we should change the name to Content. If this work is picked up a codemod issue should be made to address this change automatically for consumers.

### PR DESCRIPTION
**Pull request by adamviktora**
Do not merge this PR, file history is not tracked properly.
NEW PRs: 
- part 1:https://github.com/patternfly/patternfly-react/pull/10611
- part2: https://github.com/patternfly/patternfly-react/pull/10643

__________________________

**Followup issues**:
- [Codemod](https://github.com/patternfly/pf-codemods/issues/663)
- [org update](https://github.com/patternfly/patternfly-org/issues/4087)

**Questions**:
- Do we need the `ContentVariants` enum? I lean towards removing it as we use the typescript union type in the `component` anyways. Codemode-wise it shouldn't be a problem.

### CODE REVIEW

**Comment by nicolethoen** [comment]
🤔 

Creating a new Content component (and the noted follow up issue opened to deprecate the Text component) is raising some :red_flag: for me. Text is the MOST used component from PF according to our analytics. Is there a way we can make this change under the hood without deprecating our most used component?

**Comment by wise-king-sullyman** [comment]
@nicolethoen we could definitely keep Text around either as-is or as essentially an alternative API for this Content component (which I think would be my preferred route of the two).

That being said we'll be able to handle the Text -> Content changes for consumers via a codemod, so I'm not sure I see the same red flags 🤔 Is the objection just based on consumer familiarity with Text?

**Comment by nicolethoen** [comment]
I'm primarily concerned that we might be creating a scenario where products are going to need to make possible dozens or hundreds of manual changes to their static text and that will not go over well. If all changes here are non-breaking or something that can be handled completely by  codemods, then I'm less concerned. 

I dont' think we have the luxury to deprecate the old Text component (move it to a deprecated directory) and expect that they need to make any manual changes to move off of it.

**Comment by adamviktora** [comment]
It is all fixable via codemods (I forgot to open a codemod issue). We can do this change directly, without deprecating Text, but I think it would be good to keep the Text component on the website and announce "Here is a link to the new Content component".

**Review by kmcfaul — APPROVED** [review]

**Review by wise-king-sullyman — CHANGES_REQUESTED** [review]
Looks great!

I love the test suite 🚀 but I do think we should also test that the contentH1, contentP etc classes are all appropriately applied.

**Review by tlabaj — COMMENTED** [review]
@adamviktora  are you going to open new PR or update this one?

**Comment by adamviktora** [comment]
@tlabaj I'll update this one, regarding the new tests Austin mentioned. We will see after Office hours today if we have some new feedback - working on the slides now. But about https://github.com/patternfly/patternfly-react/issues/10580 I plan to do that in a separate PR.

**Review by tlabaj — CHANGES_REQUESTED** [review]
@adamviktora I think we should update this PR or open a new on so that it modifies Text rather than introduce a new componet since that is what we are doing.

File: packages/react-core/src/components/Content/Content.tsx [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  const isList = ['ul', 'ol', 'dl'].includes(wrappingComponent);
+
+  return (
+    <Component
+      {...ouiaProps}
+      {...props}
+      data-pf-content
+      className={css(
+        componentStyles[wrappingComponent],
+        isList && styles.content,
```
Comment (mcoker) [inline packages/react-core/src/components/Content/Content.tsx:107]: As discussed in office hours, I would remove this under the assumption that if you want to be able to write plain HTML, you'll need to wrap that markup in `<Text>`.

I think the idea for an `isContentWrapper` (or whatever it's called) prop that could be applied to a list, paragraph, etc, is a great idea and would give the functionality that exists in this PR currently, though IMO that could be an enhancement for later. If we add that, I'd like some time to test it and make sure it works as intended.

**Review by mcoker — CHANGES_REQUESTED** [review]
LGTM, just one comment around what we chatted about in office hours.

**Comment by adamviktora** [comment]
Closing this PR, because the change will be handled with 2 new PRs: https://github.com/patternfly/patternfly-react/pull/10611, https://github.com/patternfly/patternfly-react/pull/10643

**Pull request closed without merging**